### ISSUE TITLE
ActionList: Needs better guidelines on not using single select items

### ISSUE THREAD

**Issue by maximedegreve** [primer/design#260]
I found this pattern where a single selection item is used together with a confirm button.

<img width="448" alt="Screenshot 2022-04-13 at 11 27 38" src="https://user-images.githubusercontent.com/980622/163158813-19bffb99-ffd3-4539-b93b-dd39e10a6307.png">

We should provide better guidelines and examples on what the best alternatives are to this.

**Concern with this pattern**
Check marks (single selection ActionList items) are meant for final actions as they don't provide the feeling of confirming your selection.

**Things that come to my mind that could use clarification:**
- If the above design pattern is acceptable I'm assuming it should have a Cancel button as well.
- In [ActionList guidelines](https://primer.style/design/components/action-list) the SelectPanel screenshot could be improved by showing the Cancel and Confirm button and the [example in React it links to](https://primer.style/react/SelectPanel) should be adapted to include this as well.
- [Is the pull request merge pattern a good alternative?](https://user-images.githubusercontent.com/980622/163164264-b37bad7d-ee4f-4149-9243-d5121787083a.png)
- Should single selections be allowed to have a confirm button? In this case I can see why a confirm button can be useful.
- For [multiple selections](https://user-images.githubusercontent.com/2313998/158658672-8a503a7c-3857-4929-a77f-c975153e366c.png) we use checkboxes so I'm assuming here we would do the same and move to radio buttons if a confirmation is needed.
- In this case should a dialog be used instead to confirm this action? At what point should you pick a Dialog over an Overlay.

**Comment by asiermartinez** [comment]
Posting an update on a new proposal for single select options that need confirmation. This pattern can feel redundant to the "Merge pull request" button, but we've always hear concerns about that component and interaction. 

This proposal is based on the multi selection component/pattern with explicit saving. ([Docs for reference](https://primer.style/design/ui-patterns/saving#explicit-saving))

<img width="1736" alt="overlay" src="https://user-images.githubusercontent.com/150282/163223114-078e4732-dfce-466f-81b2-7938c8dd7002.png">

[Figma with the original designs](https://www.figma.com/file/eG2GQueVaKMXuWTM7i1923/Untitled?node-id=0%3A10588)

**Comment by maximedegreve** [comment]
This is now implemented.

https://primer.style/react/SelectPanel
